compose/src/main/java/com/bottlerocketstudios/compose/snippets/snippetDetails/snippetDetailsWidgets/SnippetDetailsFileCard.kt

```diff
@@ -0,0 +1,123 @@
+package com.bottlerocketstudios.compose.snippets.snippetDetails.snippetDetailsWidgets
+
+import androidx.compose.animation.animateContentSize
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Card
+import androidx.compose.material.Divider
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.OutlinedButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.bottlerocketstudios.brarchitecture.domain.models.SnippetDetailsFile
+import com.bottlerocketstudios.compose.R
+import com.bottlerocketstudios.compose.resources.Colors
+import com.bottlerocketstudios.compose.resources.Dimens
+import com.bottlerocketstudios.compose.resources.brown_grey_three
+import com.bottlerocketstudios.compose.resources.typography
+import com.bottlerocketstudios.compose.snippets.snippetDetails.returnMockSnippetDetails
+import com.bottlerocketstudios.compose.util.convertToImageBitmap
+import com.bottlerocketstudios.compose.widgets.IconText
+import com.bottlerocketstudios.launchpad.compose.light
+import com.bottlerocketstudios.launchpad.compose.normal
+
+@Composable
+fun SnippetDetailsFilesCard(file: SnippetDetailsFile?) {
```

Comment:
@MMurrayAtBR If file is null, should anything render onscreen (I'm leaning towards it not)? I'd recommend making this argument non-null.